tasks/jira.py

```diff
@@ -0,0 +1,127 @@
+from datadog_api_client.v2.model.ci_app_compute import CIAppCompute
+from datadog_api_client.v2.model.ci_app_compute_type import CIAppComputeType
+from datadog_api_client.v2.model.ci_app_query_options import CIAppQueryOptions
+from datadog_api_client.v2.model.ci_app_sort_order import CIAppSortOrder
+from datadog_api_client.v2.model.ci_app_tests_aggregate_request import CIAppTestsAggregateRequest
+from datadog_api_client.v2.model.ci_app_tests_group_by import CIAppTestsGroupBy
+from datadog_api_client.v2.model.ci_app_tests_query_filter import CIAppTestsQueryFilter
+from invoke import task
+
+
+def get_failing_tests_names() -> set[str]:
+    """
+    Returns the names of the failing tests for the last 28 days
+    """
+
+    print('Getting failing tests for the last 28 days')
+    body = CIAppTestsAggregateRequest(
+        compute=[
+            CIAppCompute(
+                aggregation=CIAppAggregationFunction.COUNT,
+                metric="@test.full_name",
+                type=CIAppComputeType.TOTAL,
+            ),
+        ],
+        filter=CIAppTestsQueryFilter(
+            _from="now-28d",
+            query="@test.service:datadog-agent @git.branch:main @test.status:fail",
+            to="now",
+        ),
+        group_by=[
+            CIAppTestsGroupBy(
+                facet="@test.full_name",
+                limit=10000,
+                sort=CIAppAggregateSort(
+                    order=CIAppSortOrder.DESCENDING,
+                ),
+                total=False,
+            ),
+        ],
+        options=CIAppQueryOptions(
+            timezone="GMT",
+        ),
+    )
+
+    configuration = Configuration()
+    with ApiClient(configuration) as api_client:
+        api_instance = CIVisibilityTestsApi(api_client)
+        response = api_instance.aggregate_ci_app_test_events(body=body)
+        result = response['data']['buckets']
+        tests = {row['by']['@test.full_name'].removeprefix('github.com/DataDog/datadog-agent/') for row in result}
+
+        return tests
+
+
+def get_jira():
+    username = os.environ['ATLASSIAN_USERNAME']
+    password = os.environ['ATLASSIAN_PASSWORD']
+    jira = Jira(url="https://datadoghq.atlassian.net", username=username, password=password)
+
+    return jira
+
+
+def close_failing_test_issue(jira, issue_key: str, verbose_test: str):
+    print('Marking as done issue', issue_key, 'for test', verbose_test)
+
+    jira.issue_add_comment(issue_key, 'Marking this issue as done since the test is not failing anymore')
+    jira.issue_transition(issue_key, 'Done')
+
+
+@task
+def close_failing_tests_stale_issues(_):
+    """
+    Will mark as done all issues created by the [failed parent tests workflow](https://app.datadoghq.com/workflow/62670e82-8416-459b-bf74-9367b8a69277) that are stale.
+    Stale is an issue:
+    - In the "To Do" section of a project
+    - Where the test has not failed since 28 days
```

Comment:
🧌 [why selecting 28 days](https://en.wikipedia.org/wiki/28_Days_Later) 😄

Reply:
This is to take 4 weeks, so 2 sprints. Considered this as a relevant time for periodic flakes to come up, and for folks to fix their tests. The rational is: if over the past 4 weeks there was no new instance of failures, this means that either the team fixed it because they spotted it out of the workflow, and did not update the ticket, or the issue is unfrequent enough to not be considered an impacting flake.

28 days is better than 30 days as it covers for weeks rather than days in a month